src/sparsezoo/model/utils.py

```diff
@@ -123,23 +112,53 @@ def load_files_from_stub(
     if isinstance(stub, str):
         stub, params = parse_zoo_stub(stub=stub, valid_params=valid_params)
     _LOGGER.debug(f"load_files_from_stub: loading files from {stub}")
-    response = download_get_request(
-        args=stub,
-        force_token_refresh=force_token_refresh,
+
+    arguments = get_model_metadata_from_stub(stub)
+
+    api = GraphQLAPI()
+
+    models = api.fetch(
+        operation_body="models",
+        arguments=arguments,
+        fields=["modelId", "modelOnnxSizeCompressedBytes"],
     )
 
-    # piece of code required for backwards compatibility
-    model_response = response.get("model", {})
-    files = model_response.get("files", [])
+    model_id = models[0]["model_id"]
+
+    files = api.fetch(
+        operation_body="files",
+        arguments={"model_id": model_id},
+    )
+    include_file_download_url(files)
     files = restructure_request_json(request_json=files)
-    compressed_file_size = _get_compressed_size(files=files)
-    model_id = model_response.get("model_id")
+
     if params is not None:
         files = filter_files(files=files, params=params)
 
-    model_results = model_response.get("results")
-    validation_results = _parse_validation_metrics(model_results_response=model_results)
-    return files, model_id, params, validation_results, compressed_file_size
+    training_results = api.fetch(
+        operation_body="training_results",
+        arguments={"model_id": model_id},
+    )
+
+    benchmark_results = api.fetch(
```

Comment:
Are we running through this flow for each search result or only when downloading a specific model? In either case, I'd think we could get this same data in a single request instead of in N requests. I appreciate that it's harder to parse that way and harder to have a generic query that way, but I guess I'm interested to know what is motivating the choice to make several calls?

Reply:
Yes thats true. Right now we are using a simple way.  
1. we can make 1 request and save it in cache, but will need to made code changes. 
2. For retrieving multiple metadata, we can add fields in the models endpoint, will need to add some engineering in the `arguments` that currently takes a list. Need it to be something like `modelId files { files_field1, ... }` so will need to change the argument type from list to something like dict for `class QueryParser`